fixtures/coverall/src/coverall.udl

```diff
@@ -47,4 +47,10 @@ interface Coveralls {
     boolean maybe_throw(boolean should_throw);
 
     void panic(string message);
+
+    // *** Test functions which take either `self` or other params as `Arc<Self>` ***
+
+    /// Calls `Arc::strong_count()` on the `Arc` containing `self`.
+    [ByArc]
```

Comment:
🚨🚲🏡🚨  **bikeshed alert** 🚨🏡🚲🚨

I'm not 100% happy with this name in this context - unless you already know what it is, it's hard to tell by reading just exact *what* will be dealt with "by arc". Something like `[SelfAsArc]` or `[Self=ByArc]` or similar might help on that front...but OTOH those are both kind of an eyesore...

I don't feel very strongly about it, and maybe it'll be rare enough that it doesn't really matter anyway. Just kinda thinking out loud about it.

Reply:
Yeah, I had the same response. I didn't really like the variants with `Self` in the name because I kinda pictured it might be used in contexts other than `self` (eg, once we start allowing interfaces as params etc). OTOH though, it seems unlikely we will ever support these params etc being anything other than as `Arc`, and in my WIP no such annotation was necessary for those contexts.

That said though, I guess I'm vaguely preferring `[Self=ByArc]` and if we did need annotations on params in the future, we'd reuse `ByArc` there.

(It does seem worth bike shedding this now)

Reply:
> I guess I'm vaguely preferring `[Self=ByArc]`

OTOH, I thought that before, so went looking for things other than `ByArc` that might be on `Self=` and came up blank - and thought that `[Self=ByArc]` when it's almost certain there will be no other value allowed there seemed dumb too. So I ended up paralyzed making a decision and moved on. And here we are again :)

Reply:
> (It does seem worth bike shedding this now)

Lol this is all the encouragement I need ;-)

> I guess I'm vaguely preferring [Self=ByArc] and if we did need annotations on params in the future, we'd reuse ByArc there.

I think this is where I've ended up as well - it's not *great*, but it reads more descriptively, it kind of fits with the ad-hoc convention we invented with `ByRef`, and would extend naturally to adding `ByArc` in other places. Might be the best of an uninspiring bunch. I think I'm willing to accept the weirdness of only having a single allowed value, just because of the way it reads better.

Perhaps our future selves will not have this problem because we're just reading `self: Arc<Self>` directly out of the underlying Rust source code 😁